src/codegen/doc/gi_docgen.rs

```diff
@@ -196,6 +196,36 @@ fn ns_type_to_doc(namespace: &Option<String>, type_: &str) -> String {
     }
 }
 
+fn find_virtual_method(
+    type_: Option<&str>,
+    namespace: Option<&str>,
+    name: &str,
+    env: &Env,
+    in_type: Option<(&TypeId, Option<LocationInObject>)>,
+) -> Option<String> {
+    find_method_or_function(
+        env,
+        in_type,
+        |f| {
+            f.name == mangle_keywords(name)
+                && namespace.as_ref().map_or(f.ns_id == MAIN_NAMESPACE, |n| {
+                    &env.library.namespaces[f.ns_id as usize].name == n
+                })
+        },
+        |o| {
+            type_.map_or(true, |t| {
+                o.name.trim_end_matches("Class") == t
```

Comment:
Isn't there something in the .gir that connects class and object?

Reply:
Also this would then need some additional strings for handling interfaces. Those are often called `XXXIface` or `XXXInterface`

Reply:
> Isn't there something in the .gir that connects class and object?

The gir files contains this information yes, but gir itself doesn't seem to do anything with it yet.

Reply:
Actually the trim here wasn't needed at all. not sure why i thought it was useful the time i wrote the code :)

Reply:
If the .gir file contains the information maybe we can use it here instead of heuristics?